### PR TITLE
PR implements batching

### PR DESCRIPTION
**Pull request by karlovnv**
Take a look at issue #146

### CODE REVIEW

File: src/progaudi.tarantool/NetworkStreamPhysicalConnection.cs [hunk 1]
```diff
@@ -46,6 +46,15 @@ public async Task Connect(ClientOptions options)
             {
                 NoDelay = true
             };
+
+            if(options.ConfigureSocket != null)
+            {
+                options.ConfigureSocket(_socket);
+            }
+
+            _socket.SendBufferSize = _socket.SendBufferSize * 2;
```
Comment (karlovnv) [inline src/progaudi.tarantool/NetworkStreamPhysicalConnection.cs]: I'll remove this hardcode

File: src/progaudi.tarantool/RequestWriter.cs [hunk 2]
```diff
@@ -79,56 +85,88 @@ public void Dispose()
...
-            }
+                result = default;
+                return false;
+             }
 
-            Tuple<ArraySegment<byte>, ArraySegment<byte>> request;
+            ArraySegment<byte> request;
             var count = 0;
-            while ((request = GetRequest()) != null)
+            UInt64 length = 0;
```
Comment (karlovnv) [inline src/progaudi.tarantool/RequestWriter.cs]: This is the only place where var is not recommended.

**Comment by karlovnv** [comment]
@aensidhe I've finished with working on your comment, please take a look

File: src/progaudi.tarantool/RequestWriter.cs (hunk 2 above)
Comment (aensithe) [reply src/progaudi.tarantool/RequestWriter.cs:140]: @karlovnv Do you support Russia?

**Pull request merged**